### PR TITLE
Extend TLT chip_sw_lc_ctrl_program_error

### PR DESCRIPTION
**Pull request by gdessouky**
This extends the `chip_sw_lc_ctrl_program_error` test to align with the `chip_sw_all_escalation_resets` test for `kTopEarlgreyAlertIdLcCtrlFatalProgError`. This should close issue #17955 (and last task item in issue #15649).

### CODE REVIEW

**Review by cindychip — APPROVED** [review]
LGTM with one question related to a comment.
Could you also help update the testplan?

File: sw/device/tests/sim_dv/lc_ctrl_program_error.c [hunk 1]
```diff
@@ -48,7 +72,196 @@ static const dif_alert_handler_class_config_t kConfigProfiles[] = {{
...
+  kCounterReset,
+  // Counter for regular interrupts.
+  kCounterInterrupt,
+  // Counter for NMIs.
+  kCounterNmi,
+};
+
+static volatile uint32_t reset_count;
+
+// This TODO comment
```
[the remark this answers is not in the document]
Comment (gdessouky) [reply sw/device/tests/sim_dv/lc_ctrl_program_error.c:121]: Oh, thanks for catching this! Will fix this and update the testplan as well.

**Comment by matutem** [comment]
@gdessouky It seems this is a copy/paste of some code in the all_escalation_reset test. Could you try to share this code instead? Either putting code in some <unit>testutils.c or just placing some code under ..._impl.[ch] files and link that code in both tests? A milder form of [clkmgr_external_clk_src_for_sw_impl.c](https://github.com/lowRISC/opentitan/blob/master/sw/device/tests/clkmgr_external_clk_src_for_sw_impl.c)

**Comment by gdessouky** [comment]
> @gdessouky It seems this is a copy/paste of some code in the all_escalation_reset test. Could you try to share this code instead? Either putting code in some testutils.c or just placing some code under ..._impl.[ch] files and link that code in both tests? A milder form of [clkmgr_external_clk_src_for_sw_impl.c](https://github.com/lowRISC/opentitan/blob/master/sw/device/tests/clkmgr_external_clk_src_for_sw_impl.c)

Thanks, yes that makes sense! I will merge this for now and work on the refactoring/sharing in a follow-up after quiet period.

**Pull request merged**